server/src/main/java/com/objectcomputing/checkins/services/checkindocument/CheckinDocumentController.java

```diff
@@ -24,18 +22,32 @@
 import io.micronaut.http.hateoas.JsonError;
 import io.micronaut.http.hateoas.Link;
 import io.micronaut.security.annotation.Secured;
-import io.micronaut.security.rules.SecurityRule;
 import io.swagger.v3.oas.annotations.tags.Tag;
 
+import java.util.concurrent.ExecutorService;
+import io.netty.channel.EventLoopGroup;
+import io.reactivex.Single;
+import io.reactivex.schedulers.Schedulers;
+
 @Controller("/services/checkin-document")
 @Secured({RoleType.Constants.ADMIN_ROLE, RoleType.Constants.PDL_ROLE})
 @Produces(MediaType.APPLICATION_JSON)
 @Tag(name = "checkin document")
-public class CheckinDocumentController {
-
-    @Inject
-    CheckinDocumentServices checkinDocumentService;
 
+    public class CheckinDocumentController {
+        
+        private final CheckinDocumentServices checkinDocumentService;
+        private final EventLoopGroup eventLoopGroup;
+        private final ExecutorService ioExecutorService;
+    
+        public CheckinDocumentController(CheckinDocumentServices checkinDocumentService,
+                                       EventLoopGroup eventLoopGroup,
+                                       ExecutorService ioExecutorService){
```

Comment:
It is important that the ExecutorService be annotated `@Named(TaskExecutors.IO)` in order to get the I/O pool otherwise we cannot guarantee this will be a separate thread pool.

Reply:
updated.